tests/test_compile_kwargs.py

```diff
@@ -0,0 +1,521 @@
+        # from_logits should have been passed and set
+        assert est.model_.loss[0].from_logits == True
+    else:
+        # at least check the name
+        assert est.model_.loss[0].__name__ == got.__name__
+
+
+@pytest.mark.parametrize(
+    "loss",
+    (
+        losses_module.BinaryCrossentropy,
+        "BinaryCrossentropy",
+        "binary_crossentropy",
+    ),
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_loss_dict(loss, n_outputs_):
+    """Tests compiling of loss when it is
+    given as an dict of losses
+    mapping to outputs.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # Test dict with global routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss={"out1": loss},
+        loss__from_logits=True,  # default is False
+    )
+    est.fit(X, y)
+    got = _class_from_strings(loss, "loss")
+    if isclass(got):
+        # from_logits should have been passed and set
+        assert est.model_.loss["out1"].from_logits == True
+    else:
+        # at least check the name
+        assert est.model_.loss["out1"].__name__ == got.__name__
+
+    # Test dict with key-based routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss={"out1": loss},
+        loss__from_logits=False,
+        loss__out1__from_logits=True,  # should override above
+    )
+    est.fit(X, y)
+    got = _class_from_strings(loss, "loss")
+    if isclass(got):
+        # from_logits should have been passed and set
+        assert est.model_.loss["out1"].from_logits == True
+    else:
+        # at least check the name
+        assert est.model_.loss["out1"].__name__ == got.__name__
+
+
+@pytest.mark.parametrize(
+    "metrics", ["binary_accuracy", metrics_module.BinaryAccuracy]
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_metrics_single_metric_per_output(metrics, n_outputs_):
+    """Metrics without the ("name", metric, "output")
+    syntax should ignore all routed and custom options.
+
+    This tests a single metric per output.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # loss functions for each output and joined show up as metrics
+    metric_idx = 1 + (n_outputs_ if n_outputs_ > 1 else 0)
+
+    metrics_class = None
+    if isclass(metrics):
+        # Test discovery in VSCode fails if TF prints out _any_
+        # warnings during discovery
+        # (which of course it does if you try to instantiate anything)
+        metrics_class = metrics
+        metrics = metrics()
+
+    # List of metrics
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss="binary_crossentropy",
+        metrics=[metrics],
+        metrics__name="custom_name",  # should be ignored
```

Comment:
I think this carried over from a previous version. I'll remove the comment and add the assert in the test.

Reply:
> neither of these should accept routed hyperparameters.

Why isn't an error raised if invalid parameters are specified?

Reply:
This was useful to test in dev, but perhaps this is not a good test to have for regression testing. I'm open to removing those checks.

Reply:
Why isn't an exception raised instead of a warning?

Reply:
Same discussion as https://github.com/adriangb/scikeras/pull/66#issuecomment-696426673, https://github.com/adriangb/scikeras/pull/66#issuecomment-696750092 and https://github.com/adriangb/scikeras/pull/66#issuecomment-696763310.

Basically, I thought there was a common use case where users might not be able to avoid routing parameters to strings and (at the time) we were not retrieving strings from classes. Now that we are retrieving strings from classes, this can be an error instead of a warning.

Reply:
I am currently working on implementing the following two types of errors:
1) ValueError if the user passes an unknown string (`loss="msq"`).
2) TypeError if the user routes parameters to a non-callable (`loss=losses.binary_crossentropy, loss__from_logits=True`).

Reply:
:+1:

Reply:
Errors implemented along with corresponding tests.

I also cleaned up the non-error tests by removing the test cases that now produce errors, it should be clearer now what they are testing.

Also all references to `_classes_from_strings` are gone from the tests.

Reply:
Since we now have errors and no longer have the `# ingored` in the tests, can I resolve this thread @stsievert ?